### PR TITLE
Renamed transformation specific classes

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
[Orchestrator to update cases with Supplementary Evidence](https://tools.hmcts.net/jira/browse/BPS-833)

### Change description ###
Renamed transformation specific classes (CaseTransformationException, TransformationErrorResponse) to more generic (CaseClientServiceException, ClientServiceErrorResponse), moved ClientServiceErrorResponse to a higher level package.

The next PR will introduce UpdateClient class - implementing low level piece of functionality for the task.

There will be a PR to call UpdateClient from service layer

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lgonczar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/CaseClientServiceException.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/model/response/ClientServiceErrorResponse.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-scan-orchestrator/bulk-scan-orchestrator/pullRequest?prid=4519300&bid=15223214)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java [hunk 1]
```diff
@@ -145,10 +145,10 @@ public void should_throw_case_transformation_exception_when_unable_to_process_bo
         );
 
         // then
-        assertThat(throwable).isInstanceOf(CaseTransformationException.class);
+        assertThat(throwable).isInstanceOf(CaseClientServiceException.class);
 
         // and
-        CaseTransformationException exception = (CaseTransformationException) throwable;
+        CaseClientServiceException exception = (CaseClientServiceException) throwable;
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java]: On the same go we could improve this by `catchThorableOfInstance` or what this method is called. This way we could move out these 2 sections and make `then` section applicable from first assertion above following lines

P.S. same recommendation applied in entire PR

**Comment by lgonczar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/CaseClientServiceException.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/model/response/ClientServiceErrorResponse.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-scan-orchestrator/bulk-scan-orchestrator/pullRequest?prid=4519300&bid=15223214)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java (hunk 1 above)
Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java]: Done

**Comment by lgonczar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/model/response/ClientServiceErrorResponse.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/CaseClientServiceException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-scan-orchestrator/bulk-scan-orchestrator/pullRequest?prid=4519300&bid=15223214)

**Pull request merged**